lib/test.rb

```diff
@@ -413,13 +413,19 @@ def setup_formulae_deps_instances(formula, formula_name)
         return
       end
 
-      build_dependents_from_source = %w[
+      build_dependents_from_source_whitelist = %w[
         cabal-install
         ghc
         go
         ocaml
+        ocaml-findlib
+        ocaml-num
         rust
-      ].include?(formula_name)
+      ]
+      build_dependents_from_source = build_dependents_from_source_whitelist.include?(formula_name) ||
+                                     (!(@formulae & build_dependents_from_source_whitelist).empty? &&
+                                      !(formula.recursive_dependencies.map { |d| d.to_formula.full_name } &
+                                        build_dependents_from_source_whitelist).empty?)
```

Comment:
Would be good to split these into multiple variables (maybe even make the whole thing a function).